src/node/utxo_snapshot.h

```diff
@@ -99,6 +99,9 @@ class SnapshotMetadata
         }
 
         s >> m_base_blockheight;
+        if (m_base_blockheight > static_cast<uint32_t>(std::numeric_limits<int>::max())) {
```

Comment:
51f197bd84916c494e9250926776b9efc3225100 maybe use `int32_t` instead of `int` for clarity?

```cpp
std::numeric_limits<int32_t>::max())
```